### PR TITLE
[media] Display visit-specific instrument list

### PR DESCRIPTION
**Pull request by ridz1208**

### CODE REVIEW

**Comment by ridz1208** [comment]
@alisterdev

File: modules/media/js/uploadForm.js [hunk 1]
```diff
@@ -361,6 +361,12 @@ var MediaUploadForm = React.createClass({
       this.state.Data.visits = this.state.Data.sessionData[value].visits;
       this.state.Data.sites = this.state.Data.sessionData[value].sites;
     }
+    var pid = this.state.formData.pscid;
```
Comment (alisterdev) [inline modules/media/js/uploadForm.js]: pid = process id?

File: modules/media/jsx/uploadForm.js [hunk 2]
```diff
@@ -344,6 +344,12 @@ var MediaUploadForm = React.createClass({
       this.state.Data.visits = this.state.Data.sessionData[value].visits;
       this.state.Data.sites = this.state.Data.sessionData[value].sites;
     }
+    var pid = this.state.formData.pscid;
```
Comment (alisterdev) [inline modules/media/jsx/uploadForm.js]: name is pscid for consistency?

**Comment by ridz1208** [comment]
@alisterdev I added the pscid triggered reload

**Comment by alisterdev** [comment]
@ridz1208 Except you didn't run ESLint

File: modules/media/jsx/uploadForm.js [hunk 3]
```diff
@@ -340,9 +340,23 @@ var MediaUploadForm = React.createClass({
    */
   setFormData: function(formElement, value) {
     // Only display visits and sites available for the current pscid
+    var visitLabel=this.state.formData.visitLabel;
     if (formElement === "pscid" && value !== "") {
       this.state.Data.visits = this.state.Data.sessionData[value].visits;
       this.state.Data.sites = this.state.Data.sessionData[value].sites;
+      if (visitLabel !== null) {
```
Comment (alisterdev) [inline modules/media/jsx/uploadForm.js]: This checks for `null`, but visit label is undefined so the condition doesn't work.
You should do instead:
```js
if (visitLabel) {
  ...
} else {
  ...
}
```

File: modules/media/jsx/uploadForm.js [hunk 4]
```diff
@@ -340,9 +340,23 @@ var MediaUploadForm = React.createClass({
...
+        this.state.Data.instruments =
+            this.state.Data.sessionData[value].instruments[visitLabel];
+      } else {
+        this.state.Data.instruments =
+            this.state.Data.sessionData[value].instruments['all'];
+      }
+    }
+    var pscid = this.state.formData.pscid;
+    if (formElement === "visitLabel" &&
+        value !== "" && pscid !== null) {
```
Comment (alisterdev) [inline modules/media/jsx/uploadForm.js]: I think the if condition is less than 80 chars so it doesn't have to go on a new line

Comment (alisterdev) [inline modules/media/jsx/uploadForm.js]: Here you can also do `&& pscid` to check for null, undefined and empty string

**Comment by alisterdev** [comment]
Looks good!

**Pull request merged**